### PR TITLE
Remove WP_REST_Widget_Utils_Controller class

### LINKED ISSUE #26092: Remove WP_REST_Widget_Utils_Controller class and form-renderer method
**Describe the solution you'd like**
Remove `WP_REST_Widget_Utils_Controller` and `get_widget_form` method from `WP_REST_Widget_Types_Controller`, as rendering form is handled by the widget controller. 

**Describe alternatives you've considered**

### PR DESCRIPTION
**Pull request by spacedmonkey**
Fixed #26092

## Description
Remove WP_REST_Widget_Utils_Controller class as no longer needed. Replaced with widget type endpoint. 

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/edit-widgets/src/blocks/legacy-widget/edit/handler.js [hunk 1]
```diff
@@ -172,9 +172,9 @@ class LegacyWidgetEditHandler extends Component {
 
 		if ( widgetClass ) {
 			apiFetch( {
-				path: `/__experimental/widget-utils/form/${ encodeURIComponent(
+				path: `/wp/v2/widget-types/${ encodeURIComponent(
 					widgetClass
```
Comment (TimothyBJacobs) [inline packages/edit-widgets/src/blocks/legacy-widget/edit/handler.js:176]: This should be using the `id-base`

**Review by TimothyBJacobs — CHANGES_REQUESTED** [review]

File: packages/edit-widgets/src/blocks/legacy-widget/edit/handler.js (hunk 1 above)
Comment (spacedmonkey) [inline packages/edit-widgets/src/blocks/legacy-widget/edit/handler.js:176]:
```suggestion
					idBase
```
Like this @TimothyBJacobs ?

Comment (TimothyBJacobs) [reply packages/edit-widgets/src/blocks/legacy-widget/edit/handler.js:176]: I believe so, but we might also need to fall back to `id` for reference widgets.

Comment (TimothyBJacobs) [inline packages/edit-widgets/src/blocks/legacy-widget/edit/handler.js:176]:
```suggestion
					idBase || id
```

**Comment by spacedmonkey** [comment]
Pinging @adamziel and @noisysocks for review.

**Comment by TimothyBJacobs** [comment]
This worked in my basic tests, waiting on #26164 to land to do final confirmation.

**Review by TimothyBJacobs — APPROVED** [review]

**Pull request merged**